util/EFSOI_Utilities/src/efsoi_main.f90

```diff
@@ -0,0 +1,166 @@
+!              pressure at each grid point/vertical level.
+!           d) decomposition of horizontal grid points and observation
+!              priors to minimize load imbalance for EFSOI calcs.
+!           e) read forecast and analysis states necessary for
+!              EFSOI calculations.
+!           f) Initialize/allocate for EFSOI moist/dry/kinetic
+!              total output
+!           g) Estimate the location of observation response
+!              (i.e. advect the localization).  Default estimation
+!              approach is to multiply meridional and zonal wind
+!              by 0.75.
+!           h) Perform EFSOI calculations for all observations
+!              considered for assimilation during the EnSRF update.
+!           i) write EFSOI calculations and ancillary EFSOI
+!              information to file
+!           j) deallocate all allocatable arrays, finalize MPI.
+!
+! program history log:
+!   2018-04-30  Initial development. Adaptation of enkf_main program
+!               towards EFSOI calculation process
+!
+! usage:
+!   input files: Update to FV3 nomenclature
+!     sigfAT_YYYYMMDDHH_mem* - Ensemble of forecasts valid at advance time
+!                              (AT) hours beyond the initial time
+!     sigfAT_YYYYMMDDHH_ensmean - Ensemble mean forecast AT hours from
+!                                 initial time
+!     sigfAT+6_YYYYMMDDHH-6_ensmean - Ensemble mean forecast AT+6 hours
+!                                     from initial time minus 6 hours
+!     siganl.YYYYMMDDHH.gdas - 
+!   output files: 
+!     obimpact_YYYYMMDDHH.dat - observation impact file
+!                         
+! comments: This program is a wrapper for components needed to peform
+!           EFSOI calculations
+!
+! attributes:
+!   language: f95
+!
+!$$$
+
+ use kinds, only: r_double,i_kind
+ ! reads namelist parameters.
+ ! applying enkf namelist apparatus
+ use params, only : read_namelist,nanals
+ ! mpi functions and variables.
+ use mpisetup, only:  mpi_initialize, mpi_initialize_io, mpi_cleanup, nproc, &
+                      mpi_wtime, mpi_comm_world
+ ! model state vector 
+ use statevec_efsoi, only: read_state_efsoi, statevec_cleanup_efsoi, init_statevec_efsoi
+ ! load balancing
+ use loadbal_efsoi, only: load_balance_efsoi, loadbal_cleanup_efsoi
+ ! efsoi update
+ use efsoi, only: efsoi_update
+ ! Observation sensitivity usage
+ use enkf_obs_sensitivity, only: init_ob_sens, print_ob_sens, destroy_ob_sens, read_ob_sens
+ use loc_advection, only: loc_advection_efsoi
+ ! Scatter chunks for EFSOI
+ use scatter_chunks_efsoi, only: scatter_chunks_ob_impact
+ use omp_lib, only: omp_get_max_threads
+ ! Needed for rad fixed files 
+ use radinfo, only: init_rad, init_rad_vars 
+ 
+ implicit none
+ integer :: ierr, nth
+ real(r_double) t1,t2
+
+ ! 0. initialize MPI.
+ call mpi_initialize()
+ if (nproc==0) call w3tagb('EFSOI_CALC',2018,0319,0055,'NP25')
```

Comment:
The date is should be 2021, no?

Reply:
Should these be hardcoded at all?

https://www.nco.ncep.noaa.gov/pmb/docs/libs/w3lib/w3tagb.html

Reply:
I guess everybody else hardcodes it

Reply:
Good question.  What are the proper values to pass to `w3tagb`?   I agree that most developers probably cut-n-paste the `w3tagb `line from other codes with minimal changes.  This isn't right, but what's right?  We should ask EIB and/or NCO for their recommendation.

Reply:
Do we need this? What does it do other than print a header at execution (which is incorrect).
We have been copy/pasting stuff without really understanding the need for it for a while.

Reply:
A scan of global-workflow shows all hardcoded, with dates going back to 1999. I could swear I've seen a line in a logfile that had the actual compilation date/time.

Reply:
I would think it could be helpful if used correctly

Reply:
This is the output:
     PROGRAM EFSOI_CALC HAS BEGUN. COMPILED 2021319.55     ORG: NP25
     STARTING DATE-TIME  MAR 16,2021  16:43:52.635   75  TUE   2459290